### DISCUSSION TITLE
Which BEX type must I use to allow background reload and parsing?

### DISCUSSION THREAD

**Discussion by realtebo** [quasarframework/quasar#11972] [CLI - BEX mode] [answered]
I need to keep 3 single pages monitored

My goal is periodically reload each of the 3 pages in the background, parse a simple UL/LI tag group, and.. then i'll think what to do, probably save into local storage and/or show into a popup but not needed.

Which kind of BEX allow me to run a period reload and parse the dom of a webpage?

**Comment by smolinari** [comment, marked as answer]
@realtebo - What do you mean by "BEX type"?

Scott

**Reply by realtebo** [reply]
popup, new tab, webpage, etc..

sorry for bad terms, it's my first journey in this world

**Reply by smolinari** [reply]
So, BEX is only the UI part to creating browser extensions. You'd need to hook into the browser APIs that allows for such manipulation, "hook" being the key word there. :) Have you created a browser extension before? Did you by chance see my article on BEX? 

Scott

**Reply by realtebo** [reply]
No, this is my first time into BEX world. I know well Quasar so I am starting merging two world. Actually I understand that I can use background / background-hooks to use chrome on install event and chrome.alarm apis to periodically do things. 
About the bridge, I did not understand a single word in the tutorial.

Please, link me your article on BEX.

**Reply by smolinari** [reply]
Here you go: https://dev.to/quasar/quasar-and-browser-extension-development-getting-bexy-part-1-39bl

Scott

**Reply by realtebo** [reply]
i read all 3parts. and of course some concepts now are clearer than before. thanks.

do you know if we can use node modules into background-hooks.js? if yes, how?

**Reply by smolinari** [reply]
A browser extension is in......the browser. No node modules available. :)

Scott

**Reply by realtebo** [reply]
Well, not so right. Because we are using quasar, and quasar uses webpack, we CAN use node modules into quasar files. and *-hooks file are handled by webpack. so in theory it's possibile. of course, everything uses file system or similar cannot work.

**Reply by smolinari** [reply]
Ah...you meant using an NPM package in background-hooks. That is possible yes, for sure. 

Scott
